qiling/arch/arch.py

```diff
@@ -3,30 +3,44 @@
 # Cross Platform and Multi Architecture Advanced Binary Emulation Framework
 #
 
-from abc import ABC
-from typing import Optional
+from abc import abstractmethod
 
 from unicorn import Uc
 from unicorn.unicorn import UcContext
 from capstone import Cs
 from keystone import Ks
 
 from qiling import Qiling
+from qiling.const import QL_ENDIAN
+from .register import QlRegisterManager
 from .utils import QlArchUtils
 
-class QlArch(ABC):
+class QlArch:
+    bits: int
+
     def __init__(self, ql: Qiling):
         self.ql = ql
         self.utils = QlArchUtils(ql)
 
-        self._disasm: Optional[Cs] = None
-        self._asm: Optional[Ks] = None
+    @property
+    @abstractmethod
+    def uc(self) -> Uc:
+        """Get unicorn instance bound to arch.
+        """
+
+        pass
 
-    # ql.init_Uc - initialized unicorn engine
     @property
-    def init_uc(self) -> Uc:
-        return self.get_init_uc()
+    @abstractmethod
+    def regs(self) -> QlRegisterManager:
```

Comment:
I suggest `ql.arch.pc` and `ql.arch.sp` to replace `ql.arch.regs.arch_pc` and `ql.arch.regs.arch_sp` because the latter is too verbose. Maybe an alias?

Reply:
Possible, but depends on whether we want to allow aliases. Aliases may provide more flexibility, but also come with code duplication and additional dependecies. Even if this one may be simple to implement, it counters the _simplicity_ guideline, IMHO, as it provides more than one way to do something.

Reply:
Yes, alias is not a good way. Maybe we can eliminate `ql.arch.regs.arch_pc` and `ql.arch.regs.arch_sp` totally?

Reply:
i think in x86 16 bit there is actually a register call sp. that is the reason why we use arch_sp before this. (i think, i dont really remember)

Reply:
ARM, MIPS and RISC-V have registers named `pc` and `sp` (:
But if we put them on `QlArch`, there will be both `ql.arch.pc` and `ql.arch.regs.pc`.
No question we need a generic method to access the architectural IP/ PC, the question it makes most sense to put that accessor and how it will be called.